udata/search/fields.py

```diff
@@ -56,7 +58,20 @@ def get_value_filter(self, value):
 
 
 class TermsFacet(Facet, DSLTermsFacet):
-    pass
+
+    def add_filter(self, filter_values):
+        if not filter_values:
```

Comment:
Sorry, I misread the plural.
```python
def add_filter(self, filter_values):
    # Perform a classic AND on values wihtout OR operator
    and_values = (v for v in filter_values if OR_OPERATOR not in v)
    and_filter = super(TermsFacet, self).add_filter(and_values)
    # Build a OR query for each value with OR
    or_values = (v for v in filter_values if OR_OPERATOR in v)
    or_filters = (
        any(Q('term',  , **{field_name: v}) for v in values.split(OR_OPERATOR))
        for values in or_values
    )  # Could be wrapped in all() but we do it once and for with and_filter
    return all(and_filter, *or_filters)
```
I tried to use generators to limit memory usage (but if Python complains, just switch to lists).
Attention, its initially a `terms` operator, not a `match` and to operate an OR, it should become multiple `term`.

What changed:
- a single `terms` query for values without `OR` operator instead of a `match` for each
- preserve the initial `AND` between filter_values (ie. `t=a|b t=c|d` should be `(a OR b) AND (c OR d)` and not `a OR b OR c OR d`

It put to my attention that dataset search `organization` field (and any field storing an ID) should have `index='not_analyzed'`